tests/ExtendedBagTest.php

```diff
@@ -453,4 +458,26 @@ public function testOxumCalculationForManyHashAlogrithm()
         $this->assertCount(1, $oxums);
         $this->assertEquals('408183.2', $oxums[0]);
     }
+
+    /**
+     * Test setting of bag-size tag.
+     * @group Extended
+     * @covers ::calculateTotalFileSizeAndAmountOfFiles
+     * @covers ::convertToHumanReadable
+     * @covers ::getBagInfoByTag
+     * @covers ::update
+     * @throws \whikloj\BagItTools\BagItException
+     */
+    public function testCalculatedBagSize()
+    {
+        $this->tmpdir = $this->prepareExtendedTestBag();
+        $bag = Bag::load($this->tmpdir);
+        $this->assertTrue($bag->validate());
+        $this->assertTrue($bag->isExtended());
+        $bag->update();
+
+        $bagSize = $bag->getBagInfoByTag('bag-size');
+        $this->assertCount(1, $bagSize);
+        $this->assertEquals('398.62 KiB', $bagSize[0]);
```

Comment:
I had written almost the exact same thing you did, but Github outages delayed my PR. So can I just suggest some additions to this test to check for correct calculation on more than one attempt and scale. I had
```php
    /**
     * Test we calculate the bag statistics correctly and persist to Payload-Oxum and Bag-Size tags.
     * @group Extended
     * @covers ::update
     * @covers ::calculateBagStats
     * @covers \whikloj\BagItTools\BagUtils::bytesToHumanReadable
     */
    public function testBagSizeCalculation()
    {
        $this->tmpdir = $this->prepareExtendedTestBag();
        $bag = Bag::load($this->tmpdir);

        $bag->addFile(self::TEST_IMAGE['filename'], 'data/image1.jpg');
        $bag->update();
        $bagsize = $bag->getBagInfoByTag('bag-size');
        $this->assertCount(1, $bagsize);
        $this->assertEquals('787.53KB', $bagsize[0]);
        $oxums = $bag->getBagInfoByTag('payload-oxum');
        $this->assertCount(1, $oxums);
        $this->assertEquals('806429.3', $oxums[0]);

        $bag->addFile(self::TEST_IMAGE['filename'], 'data/subdir/image3.jpg');
        $bag->update();
        $bagsize = $bag->getBagInfoByTag('bag-size');
        $this->assertCount(1, $bagsize);
        $this->assertEquals('1.15MB', $bagsize[0]);
        $oxums = $bag->getBagInfoByTag('payload-oxum');
        $this->assertCount(1, $oxums);
        $this->assertEquals('1204675.4', $oxums[0]);
    }
```
So if you could add the extra image in twice and ensure that it gets you to `MB` correctly I would appreciate it.

Reply:
Sorry I guess you actually need to add it 3 times to get to MB

Reply:
Thanks for this example. I adjusted the test.